client/my-sites/stats/activity-log-tasklist/index.jsx

```diff
@@ -0,0 +1,290 @@
+		allPluginsToUpdate: PropTypes.arrayOf( PropTypes.object ), // Don't use. Use connected pluginsToUpdate instead.
+
+		// Connected props
+		siteName: PropTypes.string.isRequired,
+		// Plugins already updated + those with pending updates.
+		// This extends allPluginsToUpdate with the plugin update status.
+		pluginsToUpdate: PropTypes.object.isRequired,
+
+		// Localize
+		translate: PropTypes.func.isRequired,
+		updateSinglePlugin: PropTypes.func.isRequired,
+		showErrorNotice: PropTypes.func.isRequired,
+		showInfoNotice: PropTypes.func.isRequired,
+		showSuccessNotice: PropTypes.func.isRequired,
+		removeThisNotice: PropTypes.func.isRequired,
+	};
+
+	state = {
+		pluginUpdateNotice: null, // keyed by plugin id, like "hello-dolly/hello"
+		dismissedPlugins: [],
+		dismissAll: false,
+	};
+
+	dismissAll = () => {
+		// ToDo: this should update some record in the tasklist API
+		this.setState( { dismissAll: true } );
+	};
+
+	dismissPlugin = pluginSlug => {
+		// ToDo: this should update some record in the tasklist API
+		this.setState( {
+			dismissedPlugins: this.state.dismissedPlugins.concat( [ pluginSlug ] ),
+		} );
+	};
+
+	updatePlugin = ( singlePlugin = false ) => {
+		const { removeThisNotice, showInfoNotice, siteName, updateSinglePlugin } = this.props;
+
+		// Use id: "hello-dolly/hello", slug: "hello-dolly", name: "Hello Dolly", updateStatus: bool|object,
+		const updateNoticeId = get(
+			this.state.pluginUpdateNotice,
+			[ singlePlugin.slug, 'notice', 'noticeId' ],
+			null
+		);
+
+		if ( updateNoticeId ) {
+			removeThisNotice( updateNoticeId );
+		}
+
+		updateSinglePlugin( singlePlugin );
+
+		this.setState( {
+			pluginUpdateNotice: merge( this.state.pluginUpdateNotice, {
+				[ singlePlugin.slug ]: showInfoNotice(
+					PluginNotices.inProgressMessage( 'UPDATE_PLUGIN', '1 site 1 plugin', {
+						plugin: singlePlugin.name,
+						site: siteName,
+					} ),
+					{
+						showDismiss: false,
+					}
+				),
+			} ),
+		} );
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		if ( isEmpty( nextProps.pluginsToUpdate ) ) {
+			return;
+		}
+
+		const {
+			removeThisNotice,
+			showErrorNotice,
+			showSuccessNotice,
+			siteName,
+			translate,
+			pluginsToUpdate,
+		} = nextProps;
+		const newState = {};
+		const noticesToShow = {};
+		const pluginsCompleted = [];
+
+		each( Object.values( pluginsToUpdate ), plugin => {
+			const pluginSlug = plugin.slug;
+
+			if ( false === get( this.props.pluginsToUpdate, `${ pluginSlug }.updateStatus`, false ) ) {
+				return;
+			}
+
+			if (
+				get( this.props.pluginsToUpdate, `${ pluginSlug }.updateStatus.status`, false ) ===
+					get( plugin.updateStatus, 'status', false ) ||
+				isPluginUpdating( plugin )
+			) {
+				return;
+			}
+
+			const updateStatus = plugin.updateStatus;
+			const updateNoticeId = get(
+				this.state.pluginUpdateNotice,
+				[ pluginSlug, 'notice', 'noticeId' ],
+				null
```

Comment:
on the React deprecation of `componentWillReceiveProps()` I believe we would want to put things like this into `componentDidUpdate()` instead as `componentWillReceiveProps()` may actually run multiple times before rendering and that could do funky things with the notices here.

Reply:
👍